lib/src/builder/generation/declaration_parsing.dart

```diff
@@ -282,6 +282,29 @@ class ParsedDeclarations {
       );
     }
 
+    // Ensure that Component2 declarations do not use legacy lifecycle methods.
+
+    if (declarationMap[key_component2].isNotEmpty) {
+      if (declarationMap[key_component2].first is ClassDeclaration) {
+        ClassDeclaration componentObject = declarationMap[key_component2].first;
+        List<String> legacyLifecycleEvents =
+            ['componentWillReceiveProps', 'componentWillMount', 'componentWillUpdate'];
+
+        legacyLifecycleEvents.forEach((lifecycle) {
+          dynamic method = componentObject.getMethod(lifecycle);
+
+          if (method != null) {
+            error(
+                'Error within ${componentObject.name.name}. \n\nWhen using Component2, a class cannot '
+                    'use $lifecycle because React 16 has removed $lifecycle and renamed it UNSAFE_$lifecycle. '
```

Comment:
I'd like to see this error message suggest the replacement for the legacy methods.

We could pull that off by mapping the legacy lifecycle method to its replacement like so:

```dart
if (declarationMap[key_component2].isNotEmpty) {
  final firstComponent2Member = declarationMap[key_component2].first;
  if (firstComponent2Member is ClassDeclaration) {
    Map<String, String> legacyLifecycleMethodsMap = {
      'componentWillReceiveProps': 'Use getDerivedStateFromProps instead.',
      'componentWillMount': 'Use init instead.',
      'componentWillUpdate': 'Use getSnapshotBeforeUpdate instead.',
    };

    legacyLifecycleMethodsMap.forEach((methodName, helpMessage) {
      final method = firstComponent2Member.getMethod(methodName);

      if (method != null) {
        error('''Error within ${firstComponent2Member.name.name}.
        
            When using Component2, a class cannot use $method because React 16 has removed $method and renamed it UNSAFE_$method.
            
            $helpMessage
            
            See https://reactjs.org/docs/react-component.html#legacy-lifecycle-methods for additional information.   
            ''');
      }
    });
  }
}
```

Reply:
Yep, great idea. Also, thank you for improving the comment formatting. That looks much better!